bin/deploy

```diff
@@ -4,13 +4,24 @@
 #
 #   APP defaults to carnival-staging
 #
+#   Set BUILD=1 to perform a Halcyon build
+#
 ###
 set -e
 
 case "${1:-carnival-staging}" in
-  carnival-staging) git push staging master ;;
+  carnival-staging)
+    git push staging master
```

Comment:
It's largely related to the "empty commit on a branch" thing, since the empty commit will be deployed to staging but not master.

I have a strong, negative reaction to having artifacts of the deploy environment in Git history. Things like empty commits and "skipci" tags all really bug me.